Userland/Libraries/LibGfx/ICC/TagTypes.cpp

```diff
@@ -181,6 +204,209 @@ ErrorOr<NonnullRefPtr<Lut8TagData>> Lut8TagData::from_bytes(ReadonlyBytes bytes,
         move(input_tables), move(clut_values), move(output_tables)));
 }
 
+static ErrorOr<CLUTData> read_clut_data(ReadonlyBytes bytes, AdvancedLUTHeader const& header)
+{
+    // Reads a CLUT as described in ICC v4, 10.12.3 CLUT and 10.13.5 CLUT (the two sections are virtually identical).
+    if (header.offset_to_clut + sizeof(CLUTHeader) > bytes.size())
+        return Error::from_string_literal("ICC::Profile: clut out of bounds");
+
+    if (header.number_of_input_channels >= sizeof(CLUTHeader::number_of_grid_points_in_dimension))
+        return Error::from_string_literal("ICC::Profile: clut has too many input channels");
+
+    auto& clut_header = *bit_cast<CLUTHeader const*>(bytes.data() + header.offset_to_clut);
+
+    // "Number of grid points in each dimension. Only the first i entries are used, where i is the number of input channels."
+    Vector<u8, 4> number_of_grid_points_in_dimension;
+    TRY(number_of_grid_points_in_dimension.try_resize(header.number_of_input_channels));
+    for (unsigned i = 0; i < header.number_of_input_channels; ++i)
```

Comment:
using `unsigned` for loop counters seems unusual, what's wrong with `size_t`? Same below, and especially for comparisons with `sizeof` results.